CONTRIBUTING.md

```diff
@@ -0,0 +1,138 @@
+# Contributing
+
+One of the best ways to contribute is by writing templates!
+
+### What are Templates?
+
+A template is a piece of code written in a templating language called
+[Jinja](https://jinja.palletsprojects.com/en/3.0.x/). A template defines
+a function that maps an example from a dataset in the
+[HuggingFace library](https://huggingface.co/datasets) to two strings of
+text. The first is called the _prompt_ which provides all information that
+will be available to solve a task, such as the instruction and the context.
+The second piece is called the _output_, which is the desired response to the
+prompt.
+
+### Quick-Start Guide to Writing Templates
+
+1. **Set up the app.** Fork the app and set up using the
+[README](https://github.com/bigscience-workshop/promptsource/blob/main/README.md).
+1. **Select a dataset.** Go to the tracking spreadsheet
+[here](https://docs.google.com/spreadsheets/d/10SBt96nXutB49H52PV2Lvne7F1NvVr_WZLXD8_Z0JMw/)
+and find an unclaimed one. Put your name under "Who's Prompting it?" and
+mark it yellow to show it's in progress.
+1. **Examine the dataset.** Select the dataset from the dropdown in the app.
+If the dataset has subsets (subsets are not the same as splits), you can select
+which one to work on. Note that templates are subset specific. You can find
+out background information on the dataset by reading the information in the
+app. The dataset is a collection of examples, and each example is a Python
+dictionary. The sidebar will tell you the schema that each example has.
+1. **Start a new template**. Enter a name for your first template and hit "Create."
+You can always update the name later. If you want to cancel the template, select
+"Delete Template."
+1. **Write the template**. In the box labeled "Template," enter a Jinja expression.
+See the [getting started guide](#getting-started-using-jinja-to-write-templates)
+and [cookbook](#jinja-cookbook) for details on how to write templates.
+1. **Add a reference.** If your template was inspired by a paper, note the
+reference in the "Template Reference" section. You can also add a description of
+what your template does.
+1. **Save the template**. Hit the "Save" button. The output of the template
+applied to the current example will appear in the right sidebar.
+1. **Verify the template**. Check that you didn't missed any case by scrolling through a handful of examples of the prompted dataset using the "Prompted dataset viewer" mode.
+1. **Upload the template(s).** Submit a PR using the instructions
+[here](#uploading-templates).
+1. **Duplicate the template(s).** If the dataset you have chosen bear the same
+format as other datasets (for instance `MNLI` and `SNLI` have identical format),
+you can simply claim these datasets and duplicate the templates you have written to these additional datasets.
+## Getting Started Using Jinja to Write Templates
```

Comment:
One thing I don't see included here is using Python code. Not sure what the support is there or if you plan to put that in "cookbook". I could see it being necessary for datasets like SQuAD where examples aren't just "dict of features" https://huggingface.co/datasets/squad

Reply:
SQuAD may actually be a nice example for "advanced usage" since I don't really know how we'd put that in jinja format OTOH..

Reply:
The most recent discussion I recall is that things that can't be captured in Jinja will be expressed in python classes that subclass Template. Those won't be entered through the app but just programmed and committed. (Still need to write the code that merges everything.)

Reply:
That seems reasonable to me.